### ISSUE TITLE
Computed fields functions failing when specifying the arg type in argument

### ISSUE THREAD

**Issue by sobrinho** [hasura/graphql-engine#4258]
Let say that I create a table named `people` with a computed field named `data` that the function is that:

```sql
create or replace function people_data(person people)
returns text as $$
  select 'blah'
$$ language sql stable;
```

If I try to perform an insert returning that field, the query will fail:

```graphql
mutation {
  insert_people(objects:{name:"John Coltrane"}) {
    returning {
      id
      name
      data
    }
  }
}
```

![Screen Shot 2020-03-31 at 12 21 28](https://user-images.githubusercontent.com/26460/78043881-4aec5d80-734a-11ea-8d94-d9e4bdb2ede3.png)

**Comment by sobrinho** [comment]
Although it works just fine for selects:

![Screen Shot 2020-03-31 at 14 37 13](https://user-images.githubusercontent.com/26460/78057439-1df57600-735d-11ea-9554-73247f5f5d32.png)

**Comment by rakeshkky** [comment]
@sobrinho The fix already merged to master via https://github.com/hasura/graphql-engine/pull/4080. Please test against the PR preview build image `hasura/graphql-engine:pull4080-11426891`. Closing the issue now. Feel free to open it again if still the issue persists. Thank you.

**Comment by tirumaraiselvan** [comment]
This is also now released in `v1.2.0-beta.3`
